common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -48,6 +59,28 @@ public Job createJob(String resourceTypes, String url) {
         job.setProgress(0);
         job.setUser(userService.getCurrentUser());
 
+        if (contractNumber != null) {
+            contractRepository.findContractByContractNumber(contractNumber).ifPresentOrElse(contractFound -> {
+                User user = userService.getCurrentUser();
+                Sponsor userSponsor = user.getSponsor();
+                Sponsor contractSponsor = contractFound.getSponsor();
+                if (!userSponsor.getHpmsId().equals(contractSponsor.getHpmsId())) {
```

Comment:
This will not work if the contract is assigned to a "child" organization,  but the user is from a "parent" organization, which is a different Sponsor record, but still has access to that contract.